File: api/internal/logger/logger.go
package logger

import (
	"os"
	"strings"

	"github.com/rs/zerolog"

	"api/internal/config"
)

func New(cfg *config.Config) zerolog.Logger {
	level := zerolog.InfoLevel

	switch strings.ToLower(cfg.LogLevel) {

	case "debug":
		level = zerolog.DebugLevel

	case "warn":
		level = zerolog.WarnLevel

	case "error":
		level = zerolog.ErrorLevel

	}

	zerolog.SetGlobalLevel(level)

	logger := zerolog.New(os.Stdout).
		With().
		Timestamp().
		Str("application", cfg.AppName).
		Logger()

	return logger
}

File: script/src/utils/logger.ts
export class Logger {

    constructor(
        private enabled: boolean
    ) {}

    debug(...args: unknown[]) {

        if (!this.enabled) {
            return;
        }

        console.log("[TrackingSDK]", ...args);
    }

    error(...args: unknown[]) {

        console.error("[TrackingSDK]", ...args);
    }
}